cmd/tofu/provider_source.go

```diff
@@ -17,15 +18,25 @@ import (
 
 	"github.com/opentofu/opentofu/internal/addrs"
 	"github.com/opentofu/opentofu/internal/command/cliconfig"
+	"github.com/opentofu/opentofu/internal/command/cliconfig/ociauthconfig"
 	"github.com/opentofu/opentofu/internal/getproviders"
 	"github.com/opentofu/opentofu/internal/tfdiags"
 )
 
+// ociCredsPolicyBuilder is the type of a callback function that the [providerSource]
+// functions will use if any of the configured provider installation methods
+// need to interact with OCI Distribution registries.
+//
+// We represent this indirectly as a callback function so that we can skip doing
+// this work in the common case where we won't need to interact with OCI registries
+// at all.
+type ociCredsPolicyBuilder func(context.Context) (ociauthconfig.CredentialsConfigs, error)
```

Comment:
Nit: I tend to avoid the word Builder if I can, what do you think about ociCredsPolicyBuilder -> ociCredsPolicySource?

If this is too in the weeds, feel free to just resolve it :)

Reply:
Given that the term Source already has a strong meaning in this code base, you are right that we should avoid it.  Honestly, it's not a bit issue so I'm just going to mark this as resolved.